components/NavPage.qml

```diff
@@ -0,0 +1,201 @@
+import QtQuick 2.0
+import QtLocation 5.0
+import Ubuntu.Components 1.1
+import Ubuntu.Components.Popups 1.0
+import Ubuntu.Components.ListItems 1.0 as ListItem
+import QtQuick.XmlListModel 2.0
+import QtQuick.LocalStorage 2.0
+import "../models" as Models
+import "./Helpers.js" as Helpers
+
+Page {
+    id: navPage
+    title: i18n.tr("Navigate")
+    visible: false
+
+    property string navState: ""
+    property real fromlon: 0.0
```

Comment:
Aha, yes! I have no good excuse, just that it was done over several nights and I was eager. ;)